Mathlib/Topology/ContinuousFunction/CompactlySupported.lean

```diff
@@ -0,0 +1,554 @@
+  toFun f := f.comp g
+  map_mul' _ _ := rfl
+
+/-- Composition as a linear map. -/
+def compLinearMap [AddCommMonoid δ] [ContinuousAdd δ] {R : Type*} [Semiring R] [Module R δ]
+    [ContinuousConstSMul R δ] (g : β →co γ) : C_c(γ, δ) →ₗ[R] C_c(β, δ) where
+  toFun f := f.comp g
+  map_add' _ _ := rfl
+  map_smul' _ _ := rfl
+
+/-- Composition as a non-unital algebra homomorphism. -/
+def compNonUnitalAlgHom {R : Type*} [Semiring R] [NonUnitalNonAssocSemiring δ]
+    [TopologicalSemiring δ] [Module R δ] [ContinuousConstSMul R δ] (g : β →co γ) :
+    C_c(γ, δ) →ₙₐ[R] C_c(β, δ) where
+  toFun f := f.comp g
+  map_smul' _ _ := rfl
+  map_zero' := rfl
+  map_add' _ _ := rfl
+  map_mul' _ _ := rfl
+
+end CompactlySupportedContinuousMap
+
+namespace CompactlySupportedContinuousMapClass
+
+section Basic
+
+variable [Zero β] [TopologicalSpace β] [FunLike F α β] [CompactlySupportedContinuousMapClass F α β]
+
+instance : CoeTC F (CompactlySupportedContinuousMap α β) :=
+  ⟨fun f =>
+    { toFun := f
+      continuous_toFun := map_continuous f
+      hasCompactSupport' := hasCompactSupport f }⟩
+
+/-- the product of `f : F` assuming `ContinuousMapClass F α γ` and `ContinuousSMul γ β` and
+`g : C_c(α, β)` is in `C_c(α, β)` -/
+instance [Zero β] [TopologicalSpace γ] [SMulZeroClass γ β] [ContinuousSMul γ β]
+    {F : Type*} [FunLike F α γ] [ContinuousMapClass F α γ] : SMul F C_c(α, β) where
+  smul f g :=
+    ⟨⟨fun x ↦ f x • g x, (map_continuous f).smul g.continuous⟩, g.hasCompactSupport'.smul_left⟩
+
+@[simp]
+theorem coe_smulc [Zero β] [TopologicalSpace γ] [SMulZeroClass γ β] [ContinuousSMul γ β]
+    {F : Type*} [FunLike F α γ] [ContinuousMapClass F α γ] (f : F) (g : C_c(α, β)) :
+    ⇑(f • g) = fun x => f x • g x :=
+  rfl
+
+@[simp]
+theorem smulc_apply [Zero β] [TopologicalSpace γ] [SMulZeroClass γ β] [ContinuousSMul γ β]
+    {F : Type*} [FunLike F α γ] [ContinuousMapClass F α γ] (f : F) (g : C_c(α, β)) (x : α):
+    (f • g) x = f x • g x :=
+  rfl
+
+/-- A continuous function on a compact space has automatically compact support. This is not an
+instance to avoid type class loops. -/
+lemma _root_.compactlySupportedContinuousMapClass.ofCompact (G : Type*) [FunLike G α β]
+    [ContinuousMapClass G α β] [CompactSpace α] : CompactlySupportedContinuousMapClass G α β where
+  map_continuous := map_continuous
+  hasCompactSupport := by
+    intro f
+    exact HasCompactSupport.of_compactSpace f
+
+end Basic
+
+section Uniform
+
+variable [UniformSpace β] [UniformSpace γ] [Zero γ] [FunLike F β γ]
+  [CompactlySupportedContinuousMapClass F β γ]
+
+theorem uniformContinuous (f : F) : UniformContinuous (f : β → γ) :=
+  (map_continuous f).uniformContinuous_of_tendsto_cocompact
+  (HasCompactSupport.is_zero_at_infty (hasCompactSupport f))
+
+end Uniform
+
+section ZeroAtInfty
+
+open ZeroAtInfty
+
+variable [TopologicalSpace β] [TopologicalSpace γ] [Zero γ]
+variable [FunLike F β γ] [CompactlySupportedContinuousMapClass F β γ]
+
```

Comment:
Could you instead write down the instance that `F` is a `ZeroAtInftyContinuousMapClass`?

Reply:
done.